core/src/main/java/org/elasticsearch/gateway/AsyncShardFetch.java

```diff
@@ -356,29 +356,29 @@ boolean isFetching() {
         }
 
         void markAsFetching() {
-            assert fetching == false : "double marking a node as fetching";
+            assert fetching == false: "double marking a node as fetching";
             fetching = true;
         }
 
         void doneFetching(T value) {
-            assert fetching == true : "setting value but not in fetching mode";
+            assert fetching : "setting value but not in fetching mode";
             assert failure == null : "setting value when failure already set";
             this.valueSet = true;
             this.value = value;
             this.fetching = false;
         }
 
         void doneFetching(Throwable failure) {
-            assert fetching == true : "setting value but not in fetching mode";
-            assert valueSet == false : "setting failure when already set value";
+            assert fetching : "setting value but not in fetching mode";
+            assert valueSet == false: "setting failure when already set value";
             assert failure != null : "setting failure can't be null";
             this.failure = failure;
             this.fetching = false;
         }
 
         void restartFetching() {
-            assert fetching == true : "restarting fetching, but not in fetching mode";
-            assert valueSet == false : "value can't be set when restarting fetching";
+            assert fetching : "restarting fetching, but not in fetching mode";
+            assert valueSet == false: "value can't be set when restarting fetching";
```

Comment:
Minor nit: you changed the formatting. We don't really have a standard for formatting beyond line length, and even that isn't globally enforced because we had a thousand files that violated it. But you may as well add the space back before the `:` so it looks like the code above it.